sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -368,6 +368,12 @@ class SessionCatalog(
       tableName: TableIdentifier,
       parts: Seq[CatalogTablePartition],
       ignoreIfExists: Boolean): Unit = {
+    val tableMetadata = getTableMetadata(tableName)
+    // Verify if each part exactly matches the partition spec of the table definition
+    parts.foreach { p =>
```

Comment:
Hive allows it. In RDBMS, the order of partitioning columns matters. In Hive partitioning, each partition only has a static partitioning value. Anyway, let me follow what Hive did. Thanks!